addons/product/migrations/11.0.1.2/post-migration.py

```diff
@@ -34,11 +34,11 @@ def map_product_tmpl_id_to_product_id(cr):
             for p in product_rows[1:]:
                 fields = values = column_names
```

Comment:
If `len(fields) == len(values)` according to this, why are you passing in the SQL an extra `str(p[0])` column in the SELECT part?

Reply:
well, we insert the fields ```','.join(fields + ['product_id'])```, that is ```len(fields) + 1``` (product_id has been filtered out earlier in the script) and the values we select are ```','.join(values + [str(p[0])])```, that is again ```len(fields) + 1``` (we basically copy all the values except product_id which is taken from an other earlier request. And finally the last argument r[0] is the product_packaging.id. Finally the str has been added to ensure p[0] which is an integer can be concatenated with another string. Sorry I don't see mistake, while I'm very sure the previous script was not correct. Are you sure about your remark?

Reply:
Yes, sorry. I saw bad the first string replacement. You're right.